### DISCUSSION TITLE
제페토 프로필 사진을 가져오는 방법이 있을까요?

### DISCUSSION THREAD

**Discussion by jeejul1010** [naverz/zepeto-studio-kor#10] [5. Q&A]
제페토 캐릭터 이름까지는 가져왔는데 혹시 사진은 어떤 방법으로 가져올 수 있나요? 감사합니다

**Comment by crash-k** [comment]
안녕하세요.
혹시 ZepetoPlayer 객체를 사용하고 계시면 아래 메소드를 사용해보시겠어요?
```ts
public GetProfileTextureAsync($complete: System.Action$1<UnityEngine.Texture2D>):void;
```

**Comment by hyeonjin-jo** [comment]
안녕하세요

world package 1.0.6  업데이트 이후 부터 사용 가능한 유저 프로필 이미지 함수 사용예제 공유드립니다.


```typescript
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'
import {ZepetoWorldHelper} from 'ZEPETO.World'
import {Texture,Texture2D, Sprite, Rect, Vector2} from 'UnityEngine'
import {Image} from 'UnityEngine.UI'
 
export default class GetTextureExample extends ZepetoScriptBehaviour {
     
    public userId:string;
    public sampleImage:Image;
 
    Start() {
        ZepetoWorldHelper.GetProfileTexture(this.userId,(texture:Texture)=>{
 
            this.sampleImage.sprite = this.GetSprite(texture);
             
        },(error)=>{
            console.log(error);
        });
    }
 
    GetSprite(texture:Texture){
        let rect:Rect = new Rect(0, 0, texture.width, texture.height);
        return Sprite.Create(texture as Texture2D, rect, new Vector2(0.5, 0.5));
    }
}
```

**Comment by jeejul1010** [comment]
감사합니다!!

2021년 12월 9일 (목) 오후 10:08, hyeonjin-jo ***@***.***>님이 작성:

> 안녕하세요
>
> world package 1.0.6 업데이트 이후 부터 사용 가능한 유저 프로필 이미지 함수 사용예제 공유드립니다.
>
> import { ZepetoScriptBehaviour } from 'ZEPETO.Script'
> import {ZepetoWorldHelper} from 'ZEPETO.World'
> import {Texture,Texture2D, Sprite, Rect, Vector2} from 'UnityEngine'
> import {Image} from 'UnityEngine.UI'
>
>
> export default class GetTextureExample extends ZepetoScriptBehaviour {
>
>
>
>     public userId:string;
>
>     public sampleImage:Image;
>
>
>
>     Start() {
>
>         ZepetoWorldHelper.GetProfileTexture(this.userId,(texture:Texture)=>{
>
>
>
>             this.sampleImage.sprite = this.GetSprite(texture);
>
>
>
>         },(error)=>{
>
>             console.log(error);
>
>         });
>
>     }
>
>
>
>     GetSprite(texture:Texture){
>
>         let rect:Rect = new Rect(0, 0, texture.width, texture.height);
>
>         return Sprite.Create(texture as Texture2D, rect, new Vector2(0.5, 0.5));
>
>     }
> }
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/naverz/zepeto-studio-kor/discussions/10#discussioncomment-1778970>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI4WHUOFHCV7T4W6AIOEQFLUQCS53ANCNFSM5I5EJKHA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
-- 
hellokitty<3er
